app/bundles/CampaignBundle/Resources/views/Event/_generic.html.twig

```diff
@@ -11,23 +11,129 @@
        data-event="{{ event['type'] }}"
        data-event-id="{{ event['id'] }}">
 {% endif %}
+    {% if event['type'] == 'lead.adddnc' %}
+        {% set icon = 'ri-add-line' %}
+    {% elseif event['type'] == 'lead.scorecontactscompanies' %}
+        {% set icon = 'ri-add-line' %}
+    {% elseif event['type'] == 'lead.addtocompany' %}
+        {% set icon = 'ri-add-line' %}
+    {% elseif event['type'] == 'lead.changepoints' %}
```

Comment:
We have to find a way to make these event icons easily reusable. 

I would suggest to make the custom event icon classes, we could use the LESS inheritance:
```
.ri-event-lead-adddnc {
    .ri-add-line;
}
```

And then  use it in the twig:
```
<i class="ri-{{ event['type']|replace({'.' : '-'}) }}"></i>
```

Alternatively, we could explore other approaches, such as implementing a Twig function. This is just an initial idea to make the event icons reusable.

Reply:
I tried several times to implement the LESS solution, but the compiler says they're not defined

Started importing the LESS icons:
`@import "../../../CoreBundle/Assets/css/libraries/remixicon/fonts/remixicon";`

Then
```
.campaign-event-icon[data-event-type="lead.adddnc"],
.campaign-event-icon[data-event-type="lead.scorecontactscompanies"],
.campaign-event-icon[data-event-type="lead.addtocompany"] {
    .ri-add-line;
}
```

No success, because the icons file is like this:
```
.ri-24-hours-fill:before { content: "\ea01"; }
.ri-24-hours-line:before { content: "\ea02"; }
.ri-4k-fill:before { content: "\ea03"; }
.ri-4k-line:before { content: "\ea04"; }
```

Is there something I'm forgetting, perhaps?

Reply:
You're right, it won't work with the pseudo-classes. Workarounds will not be as clear as this and I would rather avoid hardcoding the charcodes.

I would suggest using a Twig extension in the `app/bundles/CampaignBundle/Twig/Extension/CampaignEventIconExtension.php`:
```
<?php

declare(strict_types=1);

namespace Mautic\CampaignBundle\Twig\Extension;

use Twig\Extension\AbstractExtension;
use Twig\TwigFunction;

final class CampaignEventIconExtension extends AbstractExtension
{

    /**
     * @see Twig_Extension::getFunctions()
     */
    public function getFunctions(): array
    {
        return [
            new TwigFunction('getCampaignEventIcon', [$this, 'getCampaignEventIcon']),
        ];
    }

    public function getCampaignEventIcon(string $eventType): string
    {
        return match ($eventType) {
            'lead.adddnc', 'lead.scorecontactscompanies', 'lead.addtocompany' => 'ri-add-line',
            'lead.changepoints', 'campaign.addremovelead', 'stage.change', 'lead.changelist', 'lead.changetags', 'lead.updatelead', 'lead.updatecompany', 'lead.changeowner' => 'ri-edit-line',
            'lead.deletecontact' => 'ri-delete-bin-line',
            'lead.removednc' => 'ri-close-line',
            'campaign.sendwebhook', 'email.send', 'email.send.to.user' => 'ri-mail-send-line',
            'message.send' => 'ri-send-plane-line',
            default => 'ri-shapes-line',
        };
    }
}

```
Check that the icons are correct, because I've generated this part.

Then you can use it in template:
```
<i class="{{ getCampaignEventIcon(event['type']) }}"></i>
```

Reply:
It worked, thanks!